### PR TITLE
Adapt to HEMTT

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Use HEMTT as build system (uses armake2)
- Add `build.bat` for quick Windows builds without command line and `build_release.bat` for release builds
- Add script `setup.bat` to download project-specific HEMTT binary
  - No Linux script as downloading an artifact requires authentication
- Swap Travis CI and Circle CI with GitHub Actions
  - _No builds yet as Actions HEMTT is too old version (`/` comment bug)_
- Add custom HEMTT versioning script
  - Update versions in files (like `make.py`) in `prebuild` and reset in `postbuild`
  - Version source is `addons/main/script_version.hpp`
- [Documentation](https://github.com/CBATeam/CBA_A3/wiki/Building-CBA-from-source-(HEMTT))

**TODO armake2/HEMTT:**
- [x] Fix subconfig not processed correctly (https://github.com/KoffeinFlummi/armake2/issues/45)
- [x] Fix preprocessor define and comment on same line (https://github.com/KoffeinFlummi/armake2/issues/24)
  - [x] Remove temporary removal of comments
- [x] Fix preprocessor comment with `/` on same line ([example](https://github.com/CBATeam/CBA_A3/blob/0464ca4a143b92c71a0f8e04c0abac649841358f/addons/jam/CfgMagazineWells.hpp#L59)) (https://github.com/synixebrett/HEMTT/commit/477e44c5ef807f1a4ba903ce00adb7682f132d49)
- [x] Folder copy on release (https://github.com/synixebrett/HEMTT/issues/88)

---

**TODO CBA Later (User):**
- [ ] Use HEMTT file-patching symlink setup instead of `setup.py`
- [ ] Add release/publish procedure (ref. https://github.com/CBATeam/CBA_A3/pull/730#issuecomment-411018642) (HEMTT script)
- ~~Use Windows Installer instead of project-specific binary and download script (https://github.com/synixebrett/HEMTT/pull/60)~~ Will stay project-specific
- [ ] Revert #1199

**TODO HEMTT Later:**
- [ ] Release HEMTT 0.8.0 and use in CI (contains `/` comment and folder copy on release)

### CODE REVIEW

File: include/a3/editor_f/Data/Scripts/dikCodes.h [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#ifndef DIK_ESCAPE
```
Comment (dedmen) [inline include/a3/editor_f/Data/Scripts/dikCodes.h]: This is the wrong file.
https://github.com/acemod/ACE3/blob/master/include/a3/ui_f/hpp/defineDIKCodes.inc
That's the correct one. The ui_f one.

**Comment by jonpas** [comment]
Release/publish procedure idea (in combination with https://github.com/synixebrett/HEMTT/issues/16#issuecomment-504011607)
```sh
$ hemtt version -i minor
$ hemtt build --release -f
$ hemtt run publish
```

Configuration to achieve above:
```toml
version = "3.12.0.{{date \"%y%m%d\"}}" # eg. "3.12.0.190515"

[scripts.publish]
show_output = true
steps = [
    "@version -i build", # Would automatically use date specifier as above
    "git commit -am \"Bump to v{{version}}\"",
 	"git push"
]
```

---

Ideally, we could also do that (might happen in HEMTT, might not):
```sh
$ hemtt version -i minor
$ hemtt run release # Update versions, commits them, build release
```
```toml
[scripts.release]
show_output = true
steps = [
    "@version -i build", # Would automatically use date specifier as above
    "git commit -am \"Bump to v{{version}}\"",
    "build --release -f", # After version commit, we want headerext to be up-to-date
 	"git push" # Push prep/bump commit
]
```

File: addons/jam/CfgMagazineWells.hpp [hunk 2]
```diff
@@ -1,71 +1,71 @@
 class CfgMagazineWells {
 
     // Rifle calibre magwells, ordered lexicographically in metric and imperial groups
-    #include "magwells_11x59R.hpp"                  // 11x59mmR Gras | 11mm Vickers
```
Comment (dedmen) [inline addons/jam/CfgMagazineWells.hpp]: Why? There was lots of work put into these comments.

File: addons/jam/magwells_3006.hpp [hunk 3]
```diff
@@ -1,5 +1,5 @@
-    class CBA_3006_BAR {};          // M1918 BAR
```
Comment (dedmen) [inline addons/jam/magwells_3006.hpp]: What changed here? line ending?

File: addons/jam/CfgMagazineWells.hpp (hunk 2 above)
Comment (Drofseh) [reply addons/jam/CfgMagazineWells.hpp]: Indeed, those comments are essential to making sure that things are compatible, not every round is called the same thing everywhere and it's important that people are able to find the correct classes/files to put their mags in.

Comment (jonpas) [reply addons/jam/CfgMagazineWells.hpp]: Read the commit title.

File: addons/jam/magwells_3006.hpp (hunk 3 above)
Comment (jonpas) [reply addons/jam/magwells_3006.hpp]: That or empty whitespace at the end of lines.

Comment (commy2) [reply addons/jam/magwells_3006.hpp]: Line ending.

**Comment by commy2** [comment]
Why does it delete comments at the end of the line? They are actually helpful.

**Comment by jonpas** [comment]
Why does it delete comments at the end of the line? They are actually helpful.

Again, read the commit title. It is temporary so I can test the rest until it gets fixed in armake2. It's even on TODO above...

**Pull request merged**